presto-native-execution/presto_cpp/main/QueryContextManager.cpp

```diff
@@ -117,7 +117,9 @@ void updateVeloxConnectorConfigs(
     auto it = connectorConfig.find("node_selection_strategy");
     if (it != connectorConfig.end()) {
       connectorConfig[connector::hive::HiveConfig::kCacheNoRetentionSession] =
-          it->second == "SOFT_AFFINITY" ? "false" : "true";
+          (it->second == "SOFT_AFFINITY" || it->second == "HARD_AFFINITY")
```

Comment:
how about `it->second != "SOFT_AFFINITY" && it->second != "HARD_AFFINITY"`?

Reply:
Changing logic to:
```
connectorConfig[connector::hive::HiveConfig::kCacheNoRetentionSession] =
    (it->second != "SOFT_AFFINITY" && it->second != "HARD_AFFINITY") ? "true" : "false";
```